pkg/server/heartbeat.go

```diff
@@ -70,49 +73,44 @@ func (hb *heartbeat) Shutdown() {
 func (hb *heartbeat) beat() {
 	hb.logger.Info("Sending heartbeat")
 
-	statusCode, err := hb.makeHeartbeatRequest()
-	if statusCode/100 == 2 {
-		hb.logger.Info("Heartbeat successful")
-	} else {
-		hb.logger.Warnf("Heartbeat request returned a non-success response code: %s", statusCode)
-	}
-
+	heartbeatResponse, err := hb.doHeartbeatRequest()
 	if err != nil {
 		hb.logger.Warnf("An error occurred while sending heartbeat: ", err)
+		return
 	}
+
+	hb.logger.Info("Heartbeat successful")
+
+	hb.logger.Info("Updating heartbeat frequency to ", heartbeatResponse.HeartBeatIntervalSeconds)
+	hb.ticker.Stop()
+	hb.ticker = time.NewTicker(time.Duration(heartbeatResponse.HeartBeatIntervalSeconds) * time.Second)
 }
 
-func (hb *heartbeat) makeHeartbeatRequest() (int, error) {
+func (hb *heartbeat) doHeartbeatRequest() (*heartbeatResponseBody, error) {
```

Comment:
Inner function is responsible for making the request and parsing the data.  The outer function decides what to do with the resulting data (in this case, updating the heartbeat interval).